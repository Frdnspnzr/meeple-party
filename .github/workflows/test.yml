name: Test
on:
  pull_request:
    branches:
      - "*"
  workflow_dispatch:
    inputs:
      update-snapshots:
        description: "Update snapshots?"
        type: boolean
permissions:
  contents: read
  pages: write
  id-token: write
jobs:
  build:
    timeout-minutes: 10
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Build app
        run: npm run build
      - name: Save build
        uses: actions/upload-artifact@v4
        with:
          name: build
          path: .next

  e2e:
    timeout-minutes: 60
    runs-on: ubuntu-latest
    needs: build
    services:
      postgres:
        image: postgres
        env:
          POSTGRES_PASSWORD: postgres
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Install Playwright browsers
        run: npx playwright install --with-deps
      - name: Load build
        uses: actions/download-artifact@v4
        with:
          name: build
          path: .next
      - name: Set up snapshot cache
        id: snapshots_cache
        uses: actions/cache@v4
        with:
          key: ${{ runner.os }}-snapshots-${{github.ref}}
          restore-keys: ${{ runner.os }}-snapshots-refs/heads/main
          path: ./tests/e2e/snapshots/**
      - name: Update snapshots
        if: ${{steps.snapshots_cache.outputs.cache-hit != 'true' || inputs.update-snapshots == 'true'}}
        run: npx playwright test --update-snapshots
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Generate database tables
        run: npm run "prisma deploy"
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
      - name: Run Playwright tests
        run: npx playwright test --max-failures=10
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
          NEXT_PUBLIC_SECRET: Testrunner
          NEXTAUTH_URL: http://127.0.0.1:3000
      - uses: actions/upload-artifact@v4
        if: always()
        with:
          name: playwright-report
          path: playwright-report/

  unittest:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Run the tests
        run: npm test
      - name: Upload coverage reports to Codecov
        uses: codecov/codecov-action@v3
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}

  publish-e2e:
    runs-on: ubuntu-latest
    needs: [e2e, build]
    if: ${{ !cancelled() && (needs.build.result=='success') }}
    steps:
      - name: Configure GitHub Pages
        uses: actions/configure-pages@v3
      - name: Load report
        uses: actions/download-artifact@v4
        with:
          name: playwright-report
          path: playwright-report/
      - name: Upload Pages Artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: playwright-report/
      - name: Deploy to GitHub Pages
        uses: actions/deploy-pages@v2
