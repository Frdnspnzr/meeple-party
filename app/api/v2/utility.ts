import { Relationship, RelationshipType } from "@/datatypes/relationship";
import {
  UserPreferences,
  UserProfile,
  defaultUserPreferences,
} from "@/datatypes/userProfile";
import {
  Prisma,
  Relationship as PrismaRelationship,
  User,
} from "@prisma/client";

export type FullPrismaRelationship = PrismaRelationship & {
  sender: User;
  recipient: User;
};

export function convertToUserProfile(
  user: User,
  friend: boolean = false
): UserProfile {
  return {
    id: user.id,
    name: user.name,
    role: user.role,
    image: user.image,
    realName: allowRealName(user, friend) ? user.realName : null,
    about: user.about,
    place: allowPlace(user, friend) ? user.place : null,
    bggName: allowBggName(user, friend) ? user.bggName : null,
  };
}

export function normalizeRelationship(
  prismaRelationship: FullPrismaRelationship,
  userId: string
): Relationship {
  return {
    profile: getProfile(prismaRelationship, userId),
    type: getRelationshipType(prismaRelationship, userId),
    lastUpdate: prismaRelationship.updatedAt,
  };
}

function allowRealName(user: User, friend: boolean): boolean {
  if (friend) {
    return true;
  }
  const preferences = getPreferences(user);
  return preferences.showRealNameInProfile;
}

function allowPlace(user: User, friend: boolean): boolean {
  if (friend) {
    return true;
  }
  const preferences = getPreferences(user);
  return preferences.showPlaceInProfile;
}

function allowBggName(user: User, friend: boolean): boolean {
  return friend;
}

function getPreferences(user: User): UserPreferences {
  return {
    ...defaultUserPreferences,
    ...(user.preferences as Prisma.JsonObject),
  };
}

function getRelationshipType(
  relationship: FullPrismaRelationship,
  userId: string
): RelationshipType {
  if (relationship.type === "FRIENDSHIP") {
    return RelationshipType.FRIENDSHIP;
  } else if (relationship.type === "FRIEND_REQUEST") {
    if (relationship.sender.id === userId) {
      return RelationshipType.FRIEND_REQUEST_SENT;
    } else {
      return RelationshipType.FRIEND_REQUEST_RECEIVED;
    }
  }
  throw new Error("Error determining relationship type");
}

function getProfile(
  relationship: FullPrismaRelationship,
  userId: string
): UserProfile {
  const profile =
    relationship.recipientId === userId
      ? relationship.sender
      : relationship.recipient;
  return convertToUserProfile(profile, relationship.type === "FRIENDSHIP");
}

/**
 * Reads the authentication token for cronjobs from the environment variable `CRON_AUTH_TOKEN`. Throws when that variable is not set.
 * @returns Authentication token.
 * @throws Throws when no token is configured.
 */
export function getCronAuthToken() {
  if (!process.env.CRON_AUTH_TOKEN) {
    throw new Error("No cron auth token configured for current environment");
  }
  return process.env.CRON_AUTH_TOKEN;
}
