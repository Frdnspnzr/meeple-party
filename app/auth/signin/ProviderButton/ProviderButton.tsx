"use client";

import classNames from "classnames";
import { ClientSafeProvider, signIn } from "next-auth/react";
import { useCallback } from "react";
import styles from "./providerbutton.module.css";

interface ProviderButtonProps {
  provider: ClientSafeProvider;
}

const ProviderButton: React.FC<ProviderButtonProps> = ({ provider }) => {
  const login = useCallback(() => {
    signIn(provider.id).catch((error) => console.error(error));
  }, [provider]);

  return (
    <div className="row mb-2 justify-content-center">
      <div className="col-md-2 d-grid">
        <button
          type="button"
          className={classNames(
            "btn",
            styles.button,
            styles[provider.id] || "btn-primary"
          )}
          onClick={login}
        >
          <>
            {getIcon(provider.id)}&nbsp;{provider.name}
          </>
        </button>
      </div>
    </div>
  );
};

export default ProviderButton;

function getIcon(id: string) {
  switch (id) {
    case "google":
      return <i className="bi bi-google"></i>;
    case "github":
      return <i className="bi bi-github"></i>;
    default:
      return <></>;
  }
}
