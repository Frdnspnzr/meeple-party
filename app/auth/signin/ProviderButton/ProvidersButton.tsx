"use client";

import { ClientSafeProvider, signIn } from "next-auth/react";

interface ProviderButtonProps {
  provider: ClientSafeProvider;
}

const ProviderButton: React.FC<ProviderButtonProps> = ({ provider }) => {
  return (
    <div className="row mb-2 justify-content-center">
      <div className="col-md-2 d-grid">
        <button
          type="button"
          className="btn btn-primary"
          onClick={() => signIn(provider.id)}
        >
          <>
            {getIcon(provider.id)}&nbsp;{provider.name}
          </>
        </button>
      </div>
    </div>
  );
};

export default ProviderButton;

function getIcon(id: string) {
  switch (id) {
    case "google":
      return <i className="bi bi-google"></i>;
    case "github":
      return <i className="bi bi-github"></i>;
      case "twitter":
        return <i className="bi bi-twitter"></i>;
    default:
      return <></>;
  }
}
