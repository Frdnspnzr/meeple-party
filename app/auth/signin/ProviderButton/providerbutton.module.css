.button {
  transition: 220ms all ease-in-out;
}

.google {
  color: #fef7fb;
  background-color: #4285f4;
}

.google:hover, .google:focus, .google:active {
  color: #fef7fb !important;
  background-color: #306bca  !important;
}

.github {
  color: #e6edf3;
  background-color: #0d1117;
}

.github:hover, .github:focus, .github:active {
  color: #eee !important;
  background-color: #10376e  !important;
}

.discord {
  color: #FFFFFF;
  background-color: #5865F2;
}

.discord:hover, .discord:focus, .discord:active {
  color: #FFFFFF !important;
  background-color: #8891f2  !important;
}