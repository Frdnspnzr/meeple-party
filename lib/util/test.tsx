import {
  generateArray,
  generateBoolean,
  generateDate,
  generateGame,
} from "@/utility/test";
import { Game as ServerGame } from "@prisma/client";
import { Collection } from "../datatypes/client/collection";
import { Game } from "../datatypes/client/game";
import { UserId } from "../datatypes/client/userProfile";

export function ObjectData({ object }: { object: object }) {
  return (
    <>
      {Object.entries(object).map(([key, value]) => (
        <div data-testid={key} key={key}>
          {JSON.stringify(value)}
        </div>
      ))}
    </>
  );
}

export function generateCollection(userId: UserId): {
  collection: Collection;
  games: Game[];
} {
  const games = generateArray(generateGame);
  const randomGames = () => pickRandom(games).map((g) => g.id);
  const collection = {
    user: userId,
    own: randomGames(),
    wantToPlay: randomGames(),
    wishlist: randomGames(),
  };

  return {
    games,
    collection,
  };
}

export function convertClientGame(game: Game): ServerGame {
  return {
    ...game,
    thumbnail: game.thumbnail || null,
    image: game.image || null,
    BGGRank: game.BGGRank || null,
    BGGRating: game.BGGRating || null,
    updatedAt: generateDate(),
  };
}

export function pickRandom<T>(arr: T[]): T[] {
  return arr.filter(generateBoolean);
}
