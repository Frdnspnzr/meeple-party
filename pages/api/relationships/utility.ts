
import { Relationship, RelationshipType } from "@/datatypes/relationship";
import { PrivateUserProfile, PublicUserProfile } from "@/datatypes/userProfile";
import { prisma } from "@/db";
import { Session } from "@auth0/nextjs-auth0";
import { UserProfile as Auth0Profile } from "@auth0/nextjs-auth0/client";
import {
  Relationship as PrismaRelationship,
  UserProfile,
} from "@prisma/client";

export type FullPrismaRelationship = PrismaRelationship & {
  sender: UserProfile;
  recipient: UserProfile;
};

export function normalizeRelationship(
  prismaRelationship: FullPrismaRelationship,
  userId: number
): Relationship {
  return {
    profile: getProfile(prismaRelationship, userId),
    type: getRelationshipType(prismaRelationship, userId),
    lastUpdate: prismaRelationship.updatedAt,
  };
}

export function getRelationshipType(
  relationship: FullPrismaRelationship,
  userId: number
): RelationshipType {
  if (relationship.type === "FRIENDSHIP") {
    return RelationshipType.FRIENDSHIP;
  } else if (relationship.type === "FRIEND_REQUEST") {
    if (relationship.sender.id === userId) {
      return RelationshipType.FRIEND_REQUEST_SENT;
    } else {
      return RelationshipType.FRIEND_REQUEST_RECEIVED;
    }
  }
  throw new Error("Error determining relationship type");
}

export function getProfile(
  relationship: FullPrismaRelationship,
  userId: number
): PrivateUserProfile | PublicUserProfile {
  const profile =
    relationship.recipientId === userId
      ? relationship.sender
      : relationship.recipient;
  if (
    relationship.type === "FRIENDSHIP" ||
    isFriendRequestReveiced(relationship, userId)
  ) {
    return convertToPrivateProfile(profile);
  } else {
    return convertToPublicProfile(profile);
  }
}

export function convertToPrivateProfile(profile: UserProfile): PrivateUserProfile {
  return {
    ...convertToPublicProfile(profile),
    realName: profile.realName,
  };
}

export function convertToPublicProfile(profile: UserProfile): PublicUserProfile {
  return {
    id: profile.id,
    name: profile.name,
    picture: profile.picture,
  };
}

export function isFriendRequestReveiced(
  relationship: FullPrismaRelationship,
  userId: number
): boolean {
  return (
    relationship.type === "FRIEND_REQUEST" &&
    relationship.recipientId === userId
  );
}

export async function getUserProfile(session: Session): Promise<UserProfile> {
  const user = session.user as Auth0Profile;
  if (!user.email) {
    throw new Error("User from Auth0 does not have an E-Mail adress");
  }
  const userProfile = await prisma.userProfile.findUnique({
    where: { email: user.email as string },
  });

  if (!userProfile) {
    throw new Error("No user profile corresponding to Auth0 user found.");
  }

  return userProfile;
}