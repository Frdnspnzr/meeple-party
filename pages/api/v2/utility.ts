import { authOptions } from "@/pages/api/auth/[...nextauth]";
import { User } from "@prisma/client";
import { NextApiRequest, NextApiResponse } from "next";
import { getServerSession } from "next-auth/next";
import { ApiError } from "./types";

type ApiHandler = (req: NextApiRequest, res: NextApiResponse) => Promise<void>;
type ApiHandlerWithUser = (
  req: NextApiRequest,
  res: NextApiResponse,
  user: User
) => Promise<void>;

export function withUser(handler: ApiHandlerWithUser): ApiHandler {
  return async (req, res) => {
    const session = await getServerSession(req, res, authOptions);
    if (!session || !session.user) {
      throw Error("Unauthorized");
    }
    return await handler(req, res, session.user);
  };
}

export function withErrorHandling(handler: ApiHandler): ApiHandler {
  return async (req, res) => {
    try {
      return await handler(req, res);
    } catch (reason) {
      res.status(401).send({ reason } as ApiError);
    }
  };
}
