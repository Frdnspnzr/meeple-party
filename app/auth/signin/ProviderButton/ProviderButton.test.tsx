import { generateString, render } from "@/lib/utility/test";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ClientSafeProvider, signIn } from "next-auth/react";
import { SearchParamsContext } from "next/dist/shared/lib/hooks-client-context.shared-runtime";
import ProviderButton from "./ProviderButton";

jest.mock("@/i18n/client");
jest.mock("next-auth/react");

describe("Link button", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it("matches snapshot", () => {
    const provider: ClientSafeProvider = {
      id: "id",
      name: "test provider",
      type: "credentials",
      signinUrl: "https://example.com/signin-url",
      callbackUrl: "https://example.com/callback-url",
    };
    const { container } = render(<ProviderButton provider={provider} />);
    expect(container).toMatchSnapshot();
  });
  test.each(["google", "github", "discord", "twitch"])(
    "matches snapshot for provider %s",
    (providerType) => {
      const provider: ClientSafeProvider = {
        id: providerType,
        name: "test provider",
        type: "oauth",
        signinUrl: "https://example.com/signin-url",
        callbackUrl: "https://example.com/callback-url",
      };
      const { container } = render(<ProviderButton provider={provider} />);
      expect(container).toMatchSnapshot();
    }
  );
  it("calls signIn", async () => {
    const user = userEvent.setup();

    const provider: ClientSafeProvider = {
      id: generateString(),
      name: generateString(),
      type: "credentials",
      signinUrl: "https://example.com/signin-url",
      callbackUrl: "https://example.com/callback-url",
    };
    const signInMock = jest.mocked(signIn).mockRejectedValue("");

    render(<ProviderButton provider={provider} />);
    const button = screen.getByRole("button");

    await user.click(button);

    expect(signInMock.mock.calls.length).toBe(1);
    expect(signInMock.mock.calls[0][0]).toBe(provider.id);
    expect(signInMock.mock.calls[0][1]).toEqual({
      callbackUrl: undefined,
    });
  });
  it("passes callback URL to signIn", async () => {
    const user = userEvent.setup();

    const provider: ClientSafeProvider = {
      id: generateString(),
      name: generateString(),
      type: "credentials",
      signinUrl: "https://example.com/signin-url",
      callbackUrl: "https://example.com/callback-url",
    };
    const callbackUrl = generateString();
    const signInMock = jest.mocked(signIn).mockRejectedValue("");
    const searchParams = new URLSearchParams();
    searchParams.append("callbackUrl", callbackUrl);

    render(
      <SearchParamsContext.Provider value={searchParams}>
        <ProviderButton provider={provider} />
      </SearchParamsContext.Provider>
    );
    const button = screen.getByRole("button");

    await user.click(button);

    expect(signInMock.mock.calls.length).toBe(1);
    expect(signInMock.mock.calls[0][0]).toBe(provider.id);
    expect(signInMock.mock.calls[0][1]).toEqual({
      callbackUrl,
    });
  });
});
