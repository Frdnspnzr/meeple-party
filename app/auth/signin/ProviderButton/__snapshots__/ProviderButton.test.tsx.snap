// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link button matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="row mb-2 justify-content-center"
  >
    <div
      class="col-md-2 d-grid"
    >
      <button
        class="btn button id"
        type="button"
      >
        <div
          class="icon"
        />
        <div
          class="text"
        >
          test provider
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Link button matches snapshot for provider discord 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="row mb-2 justify-content-center"
  >
    <div
      class="col-md-2 d-grid"
    >
      <button
        class="btn button discord"
        type="button"
      >
        <div
          class="icon"
        >
          <i
            class="bi bi-discord"
          />
        </div>
        <div
          class="text"
        >
          test provider
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Link button matches snapshot for provider github 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="row mb-2 justify-content-center"
  >
    <div
      class="col-md-2 d-grid"
    >
      <button
        class="btn button github"
        type="button"
      >
        <div
          class="icon"
        >
          <i
            class="bi bi-github"
          />
        </div>
        <div
          class="text"
        >
          test provider
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Link button matches snapshot for provider google 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="row mb-2 justify-content-center"
  >
    <div
      class="col-md-2 d-grid"
    >
      <button
        class="btn button google"
        type="button"
      >
        <div
          class="icon"
        >
          <i
            class="bi bi-google"
          />
        </div>
        <div
          class="text"
        >
          test provider
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Link button matches snapshot for provider twitch 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="row mb-2 justify-content-center"
  >
    <div
      class="col-md-2 d-grid"
    >
      <button
        class="btn button twitch"
        type="button"
      >
        <div
          class="icon"
        >
          <i
            class="bi bi-twitch"
          />
        </div>
        <div
          class="text"
        >
          test provider
        </div>
      </button>
    </div>
  </div>
</div>
`;
